src/main/java/dev/emi/trinkets/mixin/ItemStackMixin.java

```diff
@@ -49,9 +50,16 @@ public abstract class ItemStackMixin {
 	private void getTooltip(Item.TooltipContext context, PlayerEntity player, TooltipType tooltipType, CallbackInfoReturnable<List<Text>> info, @Local(ordinal = 0) List<Text> list) {
 		TrinketsApi.getTrinketComponent(player).ifPresent(comp -> {
 			ItemStack self = (ItemStack) (Object) this;
+
+			boolean hideAdditionalTooltip = self.contains(DataComponentTypes.HIDE_ADDITIONAL_TOOLTIP);
+			boolean showAttributeTooltip = self.getOrDefault(DataComponentTypes.ATTRIBUTE_MODIFIERS, AttributeModifiersComponent.DEFAULT).showInTooltip();
+			if (hideAdditionalTooltip && !showAttributeTooltip)
+				//literally nothing to do
+				return;
+
 			boolean canEquipAnywhere = true;
 			Set<SlotType> slots = Sets.newHashSet();
-			Map<SlotType, Multimap<RegistryEntry<EntityAttribute>, EntityAttributeModifier>> modifiers = Maps.newHashMap();
+			Map<MutableText, Multimap<RegistryEntry<EntityAttribute>, EntityAttributeModifier>> modifiers = Maps.newHashMap();
```

Comment:
I believe I prefer the original implementation here, what are the benefits to having mutable text be the key rather than a potentially sortable slot type?

Reply:
Explained in the original comment. This resolves your slot "collapsing" not working

Reply:
If sortability is truly a concern, the underlying text string be used.

Reply:
Actually on.. fourth? Look, my original explanation doesn't seem to be correct. The actual fix comes from the change in the equality checking below this. I'll have to take (fifth) look tomorrow. The justification I have now is 'reduced cpu usage, probably.' Also with text and/or string keys the whole duplicate false/true loop logic could be simplified drastically (I think) to a containsKey / doesn't contain key if/else